### DISCUSSION TITLE
Run NativePHP without Laravel

### DISCUSSION THREAD

**Discussion by blessedjasonmwanza** [NativePHP/laravel#141] [Ideas]
> This is a feature request

Can it be possible to use **NativePHP** as a wrapper without having to install **Laravel**? Making it possible for us to run vanilla PHP with vanilla HTML & CSS + JS. Something similar to what [PHP DESKTOP](https://github.com/cztomczak/phpdesktop) does.

It can of course be installed via composer.

If this is already possible, kindly share some documentations we can use to achieve this. Currently, only Laravel dependant NativePHP docs are available.

**Comment by sneycampos** [comment]
This was mentioned in documentation: Right now, we only support a Laravel driver, but we're already working on making it work whatever framework you're using - and even if you're not using a framework at all.

I dont think it is time to open this type of issue

**Comment by devajmeireles** [comment]
This was released today, it's in alpha and people already want miracles, hahaha

**Comment by simonhamp** [comment]
@blessedjasonmwanza I love the enthusiasm, and thanks for raising this as an issue. It's in the pipeline, but there's no solid timeframe on this yet.

I'll leave this issue open to track development on this. For others coming here, it would be great to get some insight into your use-case: are you using another framework or plain-old-PHP?

**Reply by photocsi** [reply]
Hi, is it possible to know if any progress has been made on the use of nativephp without using frameworks? Thank you

**Reply by simonhamp** [reply]
None yet as this isn't a priority

**Comment by blessedjasonmwanza** [comment]
> @blessedjasonmwanza I love the enthusiasm, and thanks for raising this as an issue. It's in the pipeline, but there's no solid timeframe on this yet.
> 
> I'll leave this issue open to track development on this. For others coming here, it would be great to get some insight into your use-case: are you using another framework or plain old PHP?

Hi @simonhamp ,

Thank you so much for your notice.

We're currently using plain old Core PHP (and I have used this approach on most projects I have worked on during the last 6-7 years) as it allows for none-Laravel experienced developers to contribute and build onto projects quickly without spending months learning how Laravel works.

I also believe having this implemented would highly increase ```NativePHP``` adoption rate for developers and institutions developing software for the African Market (Businesses in Africa love having tools installed on their machines as compared to browser-based URL platforms which they have to visit all the time).

There is so much viability in having ```NativePHP``` wrap the ```PHP``` interpreter, mini server, and/or browser engine.

NB: Thank you so much for standing up for PHP and bringing us ```NativePHP``` as an open-source product 👏🏽  🚀

**Comment by simonhamp** [comment]
I'm moving this to a discussion (now that we have them open) so that we can keep Issues for things that are receiving active development.

Edit: To be clearer, we're _not_ actively working on supporting frameworks other than Laravel at this time. Laravel provides a really solid base for us to build a lightweight API between your application and the Tauri/Electron runtime with a known toolset.

**Reply by damms005** [reply]
If I may ask, @blessedjasonmwanza, curiously, why are you considering NativePHP over PHP Desktop that you mentioned?

**Reply by simonhamp** [reply]
@damms005 I think this should be a comment to the main thread, not a reply to my comment

**Reply by damms005** [reply]
I must admin I suck at using GitHub discussion, Simon.

Hopefully the internet lords forgive me haha.

**Reply by simonhamp** [reply]
No worries! You can't be great at everything 😉
